### ISSUE TITLE
Make Ctrl-Q (Quick-doc) and Ctrl-Shift-I (definition) work on self

### ISSUE THREAD

**Issue by dvtomas** [intellij-rust/intellij-rust#3908]
Thank you so much for #3665. It would be great, if `Ctrl-Q` and `Ctrl-Shift-I` also worked with `self`, it would be very useful after landing into unknown code..

**Comment by vlad20012** [comment]
Looks like ctrl+shift+I works. 
![Peek 2019-05-26 22-20](https://user-images.githubusercontent.com/3221931/58386186-8d475e80-8004-11e9-9eb2-a0bfc7e1ea0d.gif)

**Comment by dvtomas** [comment]
Yes, you're right. For some reason I was expecting another behavior of Ctrl-Shift-I on `self`: if `self` is of type `Foo`, I was expecting Ctrl-Shift-I to do the same as if invoked on `Foo`, showing the definition of the type of `self`, not just that `self` is a parameter of the current fn. That would IMHO be much more practical - just revealing that `self` is a parameter to the current fn is kind of obvious. So, while the current behavior is more consistent with how the other parameters behave, I'd prefer it to show the definition of `Self` instead of `self`.
